### DISCUSSION TITLE
Rev files?

### DISCUSSION THREAD

**Discussion by Sukammilla** [aonez/Keka#1441] [General]
I am trying to reassemble a group of files (.rar), but I get an error. The person who sent me the files advises me to use the existing .rev files, but Keka does not seem to notice these files and always informs me that the entire archive is corrupt. I have tried extracting the individual .revs with Keka, but not much seems to happen. How do you “force” Keka to use the .rev files to recover the entire archive?

**Comment by aonez** [comment]
Hi @Sukammilla. Be sure Keka has access to the folder where all files (rar and rev parts). Just drop the first rar file and Keka should do the rest.

#### Keka - Preferences - File Access
<img width="791" alt="Screenshot 2024-05-02 at 10 21 25" src="https://github.com/aonez/Keka/assets/480163/1949f81a-e278-46a4-9386-f2ac1d36f5bd">

Here drop the folder that contains all files to be sure it has access.

**Comment by Sukammilla** [comment]
Keka has access to any folder. This is not the problem. Maybe I expressed myself wrongly.
Keka does not seem to be able to use .rev files to recover corrupted archives (fatal error!)

![Screenshot 2024-05-02 alle 11 09 56](https://github.com/aonez/Keka/assets/168714939/d3f52b00-0723-4b08-94e9-5de60069d8c1)
![Screenshot 2024-05-02 alle 11 10 06](https://github.com/aonez/Keka/assets/168714939/b7de0622-a3d9-4dee-b801-a12e4d8c7bd9)

**Reply by aonez** [reply]
Please provide the full log. Keka uses rev when needed.

**Reply by Sukammilla** [reply]
[Keka.log](https://github.com/aonez/Keka/files/15186247/Keka.log)

**Reply by Sukammilla** [reply]
Consider that there are at least 6 .rev files...

**Comment by aonez** [comment]
Can you remove or rename the part 25 and try again?

**Reply by Sukammilla** [reply]
No way....I believe that rev files are a "curse" for apple machine...

![Screenshot 2024-05-02 alle 14 28 59](https://github.com/aonez/Keka/assets/168714939/0154838d-b0ad-441a-88f6-e64c385da2db)
[Keka2.log](https://github.com/aonez/Keka/files/15188305/Keka2.log)

**Reply by aonez** [reply]
You can see here it reconstructed some volumes:

```sh
4 volumes missing
Reconstructing...
ERROR: Bad archive /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part01.rar

Renaming /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part01.rar to /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part01.rar.bad

Creating /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part01.rar

ERROR: Bad archive /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part16.rar

Renaming /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part16.rar to /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part16.rar.bad

Creating /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part16.rar

Creating /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part25.rar

Creating /Volumes/LaCie appoggio file da smistare/Nuova cartella con elementi 2/JDOWNLOADER/Thanksgiving/8985B.part31.rar
```

Probably 4 rev files for a +30 parted file is not enough.

Anyway parts 1, 16, and 25 are broken and part 31 is broken or missing.

You can try again by deleting all those parts, but at this point I think the issue is too little rev files for too many broken files.

**Reply by Sukammilla** [reply]
You have to check, because on Windows with Winrar they tell me that the archives open correctly.

**Reply by Sukammilla** [reply]
I solved with Windup for Mac.
Check your app please

**Reply by Sukammilla** [reply]
Winzip, sorry

**Reply by aonez** [reply]
Maybe Winzip is ignoring the file checksum. Can you provide a link to the files so I can test?

**Reply by Sukammilla** [reply]
Hi, I'm sorry but I deleted the archives because I was able to solve it. However if it happens again I will post the link here. Thank you and have a nice day
